src/ScottPlot/Plottable/Circle.cs

```diff
@@ -0,0 +1,133 @@
+        {
+            switch (alignment)
+            {
+                case Alignment.LowerCenter:
+                    return (pixelX - stringWidth / 2, pixelY - stringHeight);
+                case Alignment.LowerLeft:
+                    return (pixelX, pixelY - stringHeight);
+                case Alignment.LowerRight:
+                    return (pixelX - stringWidth, pixelY - stringHeight);
+                case Alignment.MiddleLeft:
+                    return (pixelX, pixelY - stringHeight / 2);
+                case Alignment.MiddleRight:
+                    return (pixelX - stringWidth, pixelY - stringHeight / 2);
+                case Alignment.UpperCenter:
+                    return (pixelX - stringWidth / 2, pixelY);
+                case Alignment.UpperLeft:
+                    return (pixelX, pixelY);
+                case Alignment.UpperRight:
+                    return (pixelX - stringWidth, pixelY);
+                case Alignment.MiddleCenter:
+                    return (pixelX - stringWidth / 2, pixelY - stringHeight / 2);
+                default:
+                    throw new InvalidEnumArgumentException("that alignment is not recognized");
+            }
+        }
+
+        public void Render(PlotDimensions dims, Bitmap bmp, bool lowQuality = false)
+        {
+            if (string.IsNullOrWhiteSpace(Label) || IsVisible == false)
+                return;
+
+            using (Graphics gfx = GDI.Graphics(bmp, dims, lowQuality))
+            using (var font = GDI.Font(Font))
+            using (Brush fillBrush = GDI.Brush(FillColor, HatchColor, HatchStyle))
+            using (var fontBrush = new SolidBrush(Font.Color))
+            using (Pen pen = GDI.Pen(LineColor, LineWidth))
+            {
+                float pixelX = dims.GetPixelX(X);
+                float pixelY = dims.GetPixelY(Y);
+                float xRad = (float)(Radius * dims.PxPerUnitX);
+                float yRad = (float)(Radius * dims.PxPerUnitY);
+                if (!AllowEllipse)
+                {
+                    xRad = Math.Max(xRad, yRad);
+                    yRad = xRad;
+                }
+                var rect = new RectangleF(pixelX - xRad, pixelY - yRad, 2 * xRad, 2 * yRad);
+
+                if (Fill)
+                    gfx.FillEllipse(fillBrush, rect);
+                gfx.DrawEllipse(pen, rect);
+
+                if (!string.IsNullOrEmpty(Label))
+                {
+                    SizeF stringSize = GDI.MeasureString(gfx, Label, font);
+
+                    if (Font.Rotation == 0)
+                        (pixelX, pixelY) = ApplyAlignmentOffset(Font.Alignment, pixelX, pixelY, stringSize.Width, stringSize.Height);
+
+                    gfx.TranslateTransform(pixelX, pixelY);
+                    gfx.RotateTransform(Font.Rotation);
+
+                    gfx.DrawString(Label, font, fontBrush, new PointF(0, 0));
```

Comment:
On plottables the `Label` attribute determines what shows up in the legend, it is not rendered as part of the plottable. I know you grabbed this section from `Plottable.Text` which somewhat breaks the rules here, so I apologize for the lack of clarity. I think if you want a plottable that can show up in the legend *and* draws text the `Label` attribute should hold the legend's text and another attribute should hold the other text.

Of course this does depend on whether you think a `Circle` should be allowed into the legend, and I would say that it should be.